lib/src/network/protocol/proxy_protocol/header.rs

```diff
@@ -130,36 +130,170 @@ pub struct HeaderV2 {
   ver_and_cmd: u8,     // protocol version and command
   family: u8,          // protocol family and address
   len: u16,            // number of following bytes part of the header
-  addr: AddrV2,
+  addr: ProxyAddr,
 }
 
 impl HeaderV2 {
-  fn new(ver_and_cmd: u8, family: u8, addr: AddrV2) -> Self {
+  pub fn new(addr_src: SocketAddr, addr_dst: SocketAddr) -> Self {
+    let addr = ProxyAddr::from(addr_src, addr_dst);
+
     HeaderV2 { 
       signature: [0x0D, 0x0A, 0x0D, 0x0A, 0x00, 0x0D, 0x0A, 0x51, 0x55, 0x49, 0x54, 0x0A],
-      ver_and_cmd,
-      family,
-      len: 0,
+      ver_and_cmd: 0x20,
+      family: get_family(&addr),
+      len: addr.len(),
       addr,
     }
   }
+
+  pub fn into_bytes(&self) -> Vec<u8> {
+    let mut header = Vec::new();
```

Comment:
this part should be allocated with `Vec::with_capacity` since we know how many bytes we will need

Reply:
Done